libc/sysdeps: add __kernel_long and __kernel_ulong

Back port "libc/sysdeps: add __kernel_long and __kernel_ulong" from
mainline crosstool-NG tree.

Comments from uClib git hash 7fef6b983456e4c529a5239ea90715050e6f4452
reproduced here:

  Linux 3.4 added __kernel_long_t and __kernel_ulong_t and various
  exported header files were updated to use these new types. Add the
  definitions for __kernel_long_t and __kernel_ulong_t to the relevant
  kernel_types.h headers.

  This change was automated with the following scriptlet

    git grep --name-only 'typedef.*__kernel_old_dev_t' \
      | xargs sed -i '/typedef.*__kernel_old_dev_t/ a\
    typedef long\t\t__kernel_long_t;\
    typedef unsigned long\t__kernel_ulong_t;'

  Whitespace in arm, avr32, hppa, sparc was then manually fixed up.

  Signed-off-by: Chris Packham <chris.packham@alliedtelesis.co.nz>

diff --git a/patches/uClibc/0.9.32.1/200-fix-kernel-3.4plus-build.patch b/patches/uClibc/0.9.32.1/200-fix-kernel-3.4plus-build.patch
new file mode 100644
index 0000000..1c8ce99
--- /dev/null
+++ b/patches/uClibc/0.9.32.1/200-fix-kernel-3.4plus-build.patch
@@ -0,0 +1,388 @@
+diff '--exclude=.git' '--exclude=*~' --recursive -U 5 --new-file uClibc-0.9.32.1-org/libc/sysdeps/linux/alpha/bits/kernel_types.h uClibc-0.9.32.1/libc/sysdeps/linux/alpha/bits/kernel_types.h
+--- uClibc-0.9.32.1-org/libc/sysdeps/linux/alpha/bits/kernel_types.h	2011-12-30 03:11:07.000000000 -0800
++++ uClibc-0.9.32.1/libc/sysdeps/linux/alpha/bits/kernel_types.h	2014-07-17 14:51:52.439540308 -0700
+@@ -31,10 +31,12 @@
+ typedef __kernel_uid_t __kernel_old_uid_t;
+ typedef __kernel_gid_t __kernel_old_gid_t;
+ typedef __kernel_uid_t __kernel_uid32_t;
+ typedef __kernel_gid_t __kernel_gid32_t;
+ typedef __kernel_dev_t __kernel_old_dev_t;
++typedef long		__kernel_long_t;
++typedef unsigned long	__kernel_ulong_t;
+ 
+ typedef struct {
+ 	int val[2];
+ } __kernel_fsid_t;
+ 
+diff '--exclude=.git' '--exclude=*~' --recursive -U 5 --new-file uClibc-0.9.32.1-org/libc/sysdeps/linux/arm/bits/kernel_types.h uClibc-0.9.32.1/libc/sysdeps/linux/arm/bits/kernel_types.h
+--- uClibc-0.9.32.1-org/libc/sysdeps/linux/arm/bits/kernel_types.h	2011-12-30 03:11:07.000000000 -0800
++++ uClibc-0.9.32.1/libc/sysdeps/linux/arm/bits/kernel_types.h	2014-07-17 14:51:52.439540308 -0700
+@@ -30,10 +30,12 @@
+ typedef unsigned int		__kernel_gid32_t;
+ typedef unsigned short		__kernel_old_uid_t;
+ typedef unsigned short		__kernel_old_gid_t;
+ typedef long long		__kernel_loff_t;
+ typedef __kernel_dev_t		__kernel_old_dev_t;
++typedef long			__kernel_long_t;
++typedef unsigned long		__kernel_ulong_t;
+ 
+ typedef struct {
+ #ifdef __USE_ALL
+ 	int val[2];
+ #else
+diff '--exclude=.git' '--exclude=*~' --recursive -U 5 --new-file uClibc-0.9.32.1-org/libc/sysdeps/linux/avr32/bits/kernel_types.h uClibc-0.9.32.1/libc/sysdeps/linux/avr32/bits/kernel_types.h
+--- uClibc-0.9.32.1-org/libc/sysdeps/linux/avr32/bits/kernel_types.h	2011-12-30 03:11:07.000000000 -0800
++++ uClibc-0.9.32.1/libc/sysdeps/linux/avr32/bits/kernel_types.h	2014-07-17 14:51:52.439540308 -0700
+@@ -37,10 +37,12 @@
+ typedef unsigned int		__kernel_uid32_t;
+ typedef unsigned int		__kernel_gid32_t;
+ typedef unsigned short		__kernel_old_uid_t;
+ typedef unsigned short		__kernel_old_gid_t;
+ typedef unsigned short		__kernel_old_dev_t;
++typedef long			__kernel_long_t;
++typedef unsigned long		__kernel_ulong_t;
+ 
+ #ifdef __GNUC__
+ typedef long long		__kernel_loff_t;
+ #endif
+ 
+diff '--exclude=.git' '--exclude=*~' --recursive -U 5 --new-file uClibc-0.9.32.1-org/libc/sysdeps/linux/bfin/bits/kernel_types.h uClibc-0.9.32.1/libc/sysdeps/linux/bfin/bits/kernel_types.h
+--- uClibc-0.9.32.1-org/libc/sysdeps/linux/bfin/bits/kernel_types.h	2011-12-30 03:11:07.000000000 -0800
++++ uClibc-0.9.32.1/libc/sysdeps/linux/bfin/bits/kernel_types.h	2014-07-17 14:51:52.439540308 -0700
+@@ -30,10 +30,12 @@
+ typedef unsigned int	__kernel_gid32_t;
+ typedef unsigned short	__kernel_old_uid_t;
+ typedef unsigned short	__kernel_old_gid_t;
+ typedef long long	__kernel_loff_t;
+ typedef __kernel_dev_t	__kernel_old_dev_t;
++typedef long		__kernel_long_t;
++typedef unsigned long	__kernel_ulong_t;
+ 
+ typedef struct {
+ #ifdef __USE_ALL
+ 	int val[2];
+ #else
+diff '--exclude=.git' '--exclude=*~' --recursive -U 5 --new-file uClibc-0.9.32.1-org/libc/sysdeps/linux/c6x/bits/kernel_types.h uClibc-0.9.32.1/libc/sysdeps/linux/c6x/bits/kernel_types.h
+--- uClibc-0.9.32.1-org/libc/sysdeps/linux/c6x/bits/kernel_types.h	2011-12-30 03:11:07.000000000 -0800
++++ uClibc-0.9.32.1/libc/sysdeps/linux/c6x/bits/kernel_types.h	2014-07-17 14:51:52.439540308 -0700
+@@ -20,10 +20,12 @@
+ typedef unsigned int	__kernel_uid32_t;
+ typedef unsigned int	__kernel_gid32_t;
+ typedef unsigned int	__kernel_old_uid_t;
+ typedef unsigned int	__kernel_old_gid_t;
+ typedef unsigned int	__kernel_old_dev_t;
++typedef long		__kernel_long_t;
++typedef unsigned long	__kernel_ulong_t;
+ typedef unsigned int	__kernel_size_t;
+ typedef int		__kernel_ssize_t;
+ typedef int		__kernel_ptrdiff_t;
+ typedef long		__kernel_off_t;
+ typedef long long	__kernel_loff_t;
+diff '--exclude=.git' '--exclude=*~' --recursive -U 5 --new-file uClibc-0.9.32.1-org/libc/sysdeps/linux/cris/bits/kernel_types.h uClibc-0.9.32.1/libc/sysdeps/linux/cris/bits/kernel_types.h
+--- uClibc-0.9.32.1-org/libc/sysdeps/linux/cris/bits/kernel_types.h	2011-12-30 03:11:07.000000000 -0800
++++ uClibc-0.9.32.1/libc/sysdeps/linux/cris/bits/kernel_types.h	2014-07-17 14:51:52.439540308 -0700
+@@ -26,10 +26,12 @@
+ typedef unsigned int    __kernel_gid32_t;
+ 
+ typedef unsigned short  __kernel_old_uid_t;
+ typedef unsigned short  __kernel_old_gid_t;
+ typedef __kernel_dev_t	__kernel_old_dev_t;
++typedef long		__kernel_long_t;
++typedef unsigned long	__kernel_ulong_t;
+ 
+ #ifdef __GNUC__
+ typedef long long	__kernel_loff_t;
+ #endif
+ 
+diff '--exclude=.git' '--exclude=*~' --recursive -U 5 --new-file uClibc-0.9.32.1-org/libc/sysdeps/linux/e1/bits/kernel_types.h uClibc-0.9.32.1/libc/sysdeps/linux/e1/bits/kernel_types.h
+--- uClibc-0.9.32.1-org/libc/sysdeps/linux/e1/bits/kernel_types.h	2011-12-30 03:11:07.000000000 -0800
++++ uClibc-0.9.32.1/libc/sysdeps/linux/e1/bits/kernel_types.h	2014-07-17 14:51:52.439540308 -0700
+@@ -29,10 +29,12 @@
+ typedef unsigned int	__kernel_uid32_t;
+ typedef unsigned int	__kernel_gid32_t;
+ typedef unsigned short	__kernel_old_uid_t;
+ typedef unsigned short	__kernel_old_gid_t;
+ typedef __kernel_dev_t	__kernel_old_dev_t;
++typedef long		__kernel_long_t;
++typedef unsigned long	__kernel_ulong_t;
+ typedef long long	__kernel_loff_t;
+ 
+ /*
+ typedef struct {
+ #ifdef __USE_ALL
+diff '--exclude=.git' '--exclude=*~' --recursive -U 5 --new-file uClibc-0.9.32.1-org/libc/sysdeps/linux/h8300/bits/kernel_types.h uClibc-0.9.32.1/libc/sysdeps/linux/h8300/bits/kernel_types.h
+--- uClibc-0.9.32.1-org/libc/sysdeps/linux/h8300/bits/kernel_types.h	2011-12-30 03:11:07.000000000 -0800
++++ uClibc-0.9.32.1/libc/sysdeps/linux/h8300/bits/kernel_types.h	2014-07-17 14:51:52.439540308 -0700
+@@ -30,10 +30,12 @@
+ typedef unsigned int	__kernel_gid32_t;
+ typedef unsigned short	__kernel_old_uid_t;
+ typedef unsigned short	__kernel_old_gid_t;
+ typedef long long	__kernel_loff_t;
+ typedef __kernel_dev_t	__kernel_old_dev_t;
++typedef long		__kernel_long_t;
++typedef unsigned long	__kernel_ulong_t;
+ 
+ typedef struct {
+ #ifdef __USE_ALL
+ 	int val[2];
+ #else
+diff '--exclude=.git' '--exclude=*~' --recursive -U 5 --new-file uClibc-0.9.32.1-org/libc/sysdeps/linux/hppa/bits/kernel_types.h uClibc-0.9.32.1/libc/sysdeps/linux/hppa/bits/kernel_types.h
+--- uClibc-0.9.32.1-org/libc/sysdeps/linux/hppa/bits/kernel_types.h	2011-12-30 03:11:07.000000000 -0800
++++ uClibc-0.9.32.1/libc/sysdeps/linux/hppa/bits/kernel_types.h	2014-07-17 14:51:52.439540308 -0700
+@@ -43,10 +43,12 @@
+ typedef long long		__kernel_loff_t;
+ typedef long long		__kernel_off64_t;
+ typedef unsigned long long	__kernel_ino64_t;
+ 
+ typedef unsigned int		__kernel_old_dev_t;
++typedef long			__kernel_long_t;
++typedef unsigned long		__kernel_ulong_t;
+ 
+ typedef struct {
+ #ifdef __USE_ALL
+ 	int	val[2];
+ #else
+diff '--exclude=.git' '--exclude=*~' --recursive -U 5 --new-file uClibc-0.9.32.1-org/libc/sysdeps/linux/i386/bits/kernel_types.h uClibc-0.9.32.1/libc/sysdeps/linux/i386/bits/kernel_types.h
+--- uClibc-0.9.32.1-org/libc/sysdeps/linux/i386/bits/kernel_types.h	2011-12-30 03:11:07.000000000 -0800
++++ uClibc-0.9.32.1/libc/sysdeps/linux/i386/bits/kernel_types.h	2014-07-17 14:51:52.443540303 -0700
+@@ -38,10 +38,12 @@
+ typedef unsigned int	__kernel_uid32_t;
+ typedef unsigned int	__kernel_gid32_t;
+ typedef unsigned short	__kernel_old_uid_t;
+ typedef unsigned short	__kernel_old_gid_t;
+ typedef __kernel_dev_t	__kernel_old_dev_t;
++typedef long		__kernel_long_t;
++typedef unsigned long	__kernel_ulong_t;
+ typedef long long	__kernel_loff_t;
+ 
+ typedef struct {
+ #ifdef __USE_ALL
+ 	int val[2];
+diff '--exclude=.git' '--exclude=*~' --recursive -U 5 --new-file uClibc-0.9.32.1-org/libc/sysdeps/linux/ia64/bits/kernel_types.h uClibc-0.9.32.1/libc/sysdeps/linux/ia64/bits/kernel_types.h
+--- uClibc-0.9.32.1-org/libc/sysdeps/linux/ia64/bits/kernel_types.h	2011-12-30 03:11:07.000000000 -0800
++++ uClibc-0.9.32.1/libc/sysdeps/linux/ia64/bits/kernel_types.h	2014-07-17 14:51:52.443540303 -0700
+@@ -50,7 +50,9 @@
+ typedef __kernel_uid_t __kernel_uid32_t;
+ typedef __kernel_gid_t __kernel_gid32_t;
+ 
+ typedef unsigned int	__kernel_dev_t;
+ typedef unsigned int	__kernel_old_dev_t;
++typedef long		__kernel_long_t;
++typedef unsigned long	__kernel_ulong_t;
+ 
+ #endif /* _ASM_IA64_POSIX_TYPES_H */
+diff '--exclude=.git' '--exclude=*~' --recursive -U 5 --new-file uClibc-0.9.32.1-org/libc/sysdeps/linux/m68k/bits/kernel_types.h uClibc-0.9.32.1/libc/sysdeps/linux/m68k/bits/kernel_types.h
+--- uClibc-0.9.32.1-org/libc/sysdeps/linux/m68k/bits/kernel_types.h	2011-12-30 03:11:07.000000000 -0800
++++ uClibc-0.9.32.1/libc/sysdeps/linux/m68k/bits/kernel_types.h	2014-07-17 14:51:52.443540303 -0700
+@@ -30,10 +30,12 @@
+ typedef unsigned int	__kernel_gid32_t;
+ typedef unsigned short	__kernel_old_uid_t;
+ typedef unsigned short	__kernel_old_gid_t;
+ typedef long long	__kernel_loff_t;
+ typedef __kernel_dev_t	__kernel_old_dev_t;
++typedef long		__kernel_long_t;
++typedef unsigned long	__kernel_ulong_t;
+ 
+ typedef struct {
+ #ifdef __USE_ALL
+ 	int val[2];
+ #else
+diff '--exclude=.git' '--exclude=*~' --recursive -U 5 --new-file uClibc-0.9.32.1-org/libc/sysdeps/linux/microblaze/bits/kernel_types.h uClibc-0.9.32.1/libc/sysdeps/linux/microblaze/bits/kernel_types.h
+--- uClibc-0.9.32.1-org/libc/sysdeps/linux/microblaze/bits/kernel_types.h	2011-12-30 03:11:07.000000000 -0800
++++ uClibc-0.9.32.1/libc/sysdeps/linux/microblaze/bits/kernel_types.h	2014-07-17 14:51:52.443540303 -0700
+@@ -42,10 +42,12 @@
+ typedef unsigned int	__kernel_gid32_t;
+ 
+ typedef unsigned int	__kernel_old_uid_t;
+ typedef unsigned int	__kernel_old_gid_t;
+ typedef unsigned int	__kernel_old_dev_t;
++typedef long		__kernel_long_t;
++typedef unsigned long	__kernel_ulong_t;
+ 
+ #ifdef __GNUC__
+ typedef long long	__kernel_loff_t;
+ #endif
+ 
+diff '--exclude=.git' '--exclude=*~' --recursive -U 5 --new-file uClibc-0.9.32.1-org/libc/sysdeps/linux/mips/bits/kernel_types.h uClibc-0.9.32.1/libc/sysdeps/linux/mips/bits/kernel_types.h
+--- uClibc-0.9.32.1-org/libc/sysdeps/linux/mips/bits/kernel_types.h	2011-12-30 03:11:07.000000000 -0800
++++ uClibc-0.9.32.1/libc/sysdeps/linux/mips/bits/kernel_types.h	2014-07-17 14:51:52.443540303 -0700
+@@ -30,10 +30,12 @@
+ typedef int		__kernel_uid32_t;
+ typedef int		__kernel_gid32_t;
+ typedef __kernel_uid_t	__kernel_old_uid_t;
+ typedef __kernel_gid_t	__kernel_old_gid_t;
+ typedef __kernel_dev_t	__kernel_old_dev_t;
++typedef long		__kernel_long_t;
++typedef unsigned long	__kernel_ulong_t;
+ typedef long long      __kernel_loff_t;
+ #else
+ typedef unsigned int	__kernel_dev_t;
+ typedef unsigned long	__kernel_ino_t;
+ typedef unsigned int	__kernel_mode_t;
+@@ -66,10 +68,12 @@
+ typedef int		__kernel_uid32_t;
+ typedef int		__kernel_gid32_t;
+ typedef __kernel_uid_t	__kernel_old_uid_t;
+ typedef __kernel_gid_t	__kernel_old_gid_t;
+ typedef __kernel_dev_t	__kernel_old_dev_t;
++typedef long		__kernel_long_t;
++typedef unsigned long	__kernel_ulong_t;
+ typedef long long      __kernel_loff_t;
+ #endif
+ 
+ 
+ typedef struct {
+diff '--exclude=.git' '--exclude=*~' --recursive -U 5 --new-file uClibc-0.9.32.1-org/libc/sysdeps/linux/nios2/bits/kernel_types.h uClibc-0.9.32.1/libc/sysdeps/linux/nios2/bits/kernel_types.h
+--- uClibc-0.9.32.1-org/libc/sysdeps/linux/nios2/bits/kernel_types.h	2011-12-30 03:11:07.000000000 -0800
++++ uClibc-0.9.32.1/libc/sysdeps/linux/nios2/bits/kernel_types.h	2014-07-17 14:51:52.443540303 -0700
+@@ -29,10 +29,12 @@
+ typedef unsigned int	__kernel_uid32_t;
+ typedef unsigned int	__kernel_gid32_t;
+ typedef unsigned short	__kernel_old_uid_t;
+ typedef unsigned short	__kernel_old_gid_t;
+ typedef unsigned short	__kernel_old_dev_t;
++typedef long		__kernel_long_t;
++typedef unsigned long	__kernel_ulong_t;
+ typedef long long	__kernel_loff_t;
+ 
+ typedef struct {
+ #ifdef __USE_ALL
+ 	int val[2];
+diff '--exclude=.git' '--exclude=*~' --recursive -U 5 --new-file uClibc-0.9.32.1-org/libc/sysdeps/linux/powerpc/bits/kernel_types.h uClibc-0.9.32.1/libc/sysdeps/linux/powerpc/bits/kernel_types.h
+--- uClibc-0.9.32.1-org/libc/sysdeps/linux/powerpc/bits/kernel_types.h	2011-12-30 03:11:07.000000000 -0800
++++ uClibc-0.9.32.1/libc/sysdeps/linux/powerpc/bits/kernel_types.h	2014-07-17 14:51:52.443540303 -0700
+@@ -34,10 +34,12 @@
+ typedef unsigned int	__kernel_uid32_t;
+ typedef unsigned int	__kernel_gid32_t;
+ typedef unsigned int	__kernel_old_uid_t;
+ typedef unsigned int	__kernel_old_gid_t;
+ typedef __kernel_dev_t	__kernel_old_dev_t;
++typedef long		__kernel_long_t;
++typedef unsigned long	__kernel_ulong_t;
+ #else
+ typedef unsigned int	__kernel_dev_t;
+ typedef unsigned int	__kernel_ino_t;
+ typedef unsigned int	__kernel_mode_t;
+ typedef unsigned short	__kernel_nlink_t;
+@@ -59,10 +61,12 @@
+ typedef unsigned int	__kernel_uid32_t;
+ typedef unsigned int	__kernel_gid32_t;
+ typedef unsigned int	__kernel_old_uid_t;
+ typedef unsigned int	__kernel_old_gid_t;
+ typedef __kernel_dev_t	__kernel_old_dev_t;
++typedef long		__kernel_long_t;
++typedef unsigned long	__kernel_ulong_t;
+ typedef long long	__kernel_loff_t;
+ #endif
+ 
+ typedef struct {
+ 	int val[2];
+diff '--exclude=.git' '--exclude=*~' --recursive -U 5 --new-file uClibc-0.9.32.1-org/libc/sysdeps/linux/sh/bits/kernel_types.h uClibc-0.9.32.1/libc/sysdeps/linux/sh/bits/kernel_types.h
+--- uClibc-0.9.32.1-org/libc/sysdeps/linux/sh/bits/kernel_types.h	2011-12-30 03:11:07.000000000 -0800
++++ uClibc-0.9.32.1/libc/sysdeps/linux/sh/bits/kernel_types.h	2014-07-17 14:51:52.443540303 -0700
+@@ -30,10 +30,12 @@
+ typedef unsigned int	__kernel_uid32_t;
+ typedef unsigned int	__kernel_gid32_t;
+ typedef unsigned short	__kernel_old_uid_t;
+ typedef unsigned short	__kernel_old_gid_t;
+ typedef __kernel_dev_t	__kernel_old_dev_t;
++typedef long		__kernel_long_t;
++typedef unsigned long	__kernel_ulong_t;
+ typedef long long	__kernel_loff_t;
+ 
+ typedef struct {
+ #ifdef __USE_ALL
+ 	int val[2];
+diff '--exclude=.git' '--exclude=*~' --recursive -U 5 --new-file uClibc-0.9.32.1-org/libc/sysdeps/linux/sh64/bits/kernel_types.h uClibc-0.9.32.1/libc/sysdeps/linux/sh64/bits/kernel_types.h
+--- uClibc-0.9.32.1-org/libc/sysdeps/linux/sh64/bits/kernel_types.h	2011-12-30 03:11:07.000000000 -0800
++++ uClibc-0.9.32.1/libc/sysdeps/linux/sh64/bits/kernel_types.h	2014-07-17 14:51:52.447540298 -0700
+@@ -41,10 +41,12 @@
+ typedef unsigned int    __kernel_uid32_t;
+ typedef unsigned int    __kernel_gid32_t;
+ typedef unsigned short  __kernel_old_uid_t;
+ typedef unsigned short  __kernel_old_gid_t;
+ typedef __kernel_dev_t	__kernel_old_dev_t;
++typedef long		__kernel_long_t;
++typedef unsigned long	__kernel_ulong_t;
+ typedef long long       __kernel_loff_t;
+ 
+ typedef struct {
+ #ifdef __USE_ALL
+ 	int val[2];
+diff '--exclude=.git' '--exclude=*~' --recursive -U 5 --new-file uClibc-0.9.32.1-org/libc/sysdeps/linux/sparc/bits/kernel_types.h uClibc-0.9.32.1/libc/sysdeps/linux/sparc/bits/kernel_types.h
+--- uClibc-0.9.32.1-org/libc/sysdeps/linux/sparc/bits/kernel_types.h	2011-12-30 03:11:07.000000000 -0800
++++ uClibc-0.9.32.1/libc/sysdeps/linux/sparc/bits/kernel_types.h	2014-07-17 14:51:52.447540298 -0700
+@@ -30,10 +30,12 @@
+ typedef unsigned short	       __kernel_uid16_t;
+ typedef unsigned short	       __kernel_gid16_t;
+ typedef __kernel_uid_t 	       __kernel_old_uid_t;
+ typedef __kernel_gid_t         __kernel_old_gid_t;
+ typedef __kernel_dev_t         __kernel_old_dev_t;
++typedef long                   __kernel_long_t;
++typedef unsigned long          __kernel_ulong_t;
+ typedef __kernel_uid_t	       __kernel_uid32_t;
+ typedef __kernel_gid_t	       __kernel_gid32_t;
+ typedef int		       __kernel_suseconds_t;
+ typedef long long              __kernel_loff_t;
+ #else
+@@ -60,10 +62,12 @@
+ typedef unsigned int	       __kernel_uid32_t;
+ typedef unsigned int	       __kernel_gid32_t;
+ typedef unsigned short	       __kernel_old_uid_t;
+ typedef unsigned short	       __kernel_old_gid_t;
+ typedef __kernel_dev_t         __kernel_old_dev_t;
++typedef long                   __kernel_long_t;
++typedef unsigned long          __kernel_ulong_t;
+ typedef long long              __kernel_loff_t;
+ #endif
+ 
+ typedef struct {
+ #ifdef __USE_ALL
+diff '--exclude=.git' '--exclude=*~' --recursive -U 5 --new-file uClibc-0.9.32.1-org/libc/sysdeps/linux/v850/bits/kernel_types.h uClibc-0.9.32.1/libc/sysdeps/linux/v850/bits/kernel_types.h
+--- uClibc-0.9.32.1-org/libc/sysdeps/linux/v850/bits/kernel_types.h	2011-12-30 03:11:07.000000000 -0800
++++ uClibc-0.9.32.1/libc/sysdeps/linux/v850/bits/kernel_types.h	2014-07-17 14:51:52.447540298 -0700
+@@ -39,10 +39,12 @@
+ typedef unsigned int	__kernel_gid32_t;
+ 
+ typedef unsigned short	__kernel_old_uid_t;
+ typedef unsigned short	__kernel_old_gid_t;
+ typedef __kernel_dev_t	__kernel_old_dev_t;
++typedef long		__kernel_long_t;
++typedef unsigned long	__kernel_ulong_t;
+ 
+ typedef struct {
+ #ifdef __USE_ALL
+ 	int val[2];
+ #else
+diff '--exclude=.git' '--exclude=*~' --recursive -U 5 --new-file uClibc-0.9.32.1-org/libc/sysdeps/linux/x86_64/bits/kernel_types.h uClibc-0.9.32.1/libc/sysdeps/linux/x86_64/bits/kernel_types.h
+--- uClibc-0.9.32.1-org/libc/sysdeps/linux/x86_64/bits/kernel_types.h	2011-12-30 03:11:07.000000000 -0800
++++ uClibc-0.9.32.1/libc/sysdeps/linux/x86_64/bits/kernel_types.h	2014-07-17 14:51:52.447540298 -0700
+@@ -38,10 +38,12 @@
+ typedef unsigned int	__kernel_uid32_t;
+ typedef unsigned int	__kernel_gid32_t;
+ typedef unsigned short	__kernel_old_uid_t;
+ typedef unsigned short	__kernel_old_gid_t;
+ typedef __kernel_dev_t	__kernel_old_dev_t;
++typedef long		__kernel_long_t;
++typedef unsigned long	__kernel_ulong_t;
+ typedef long long	__kernel_loff_t;
+ 
+ typedef struct {
+ #ifdef __USE_ALL
+ 	int val[2];
+diff '--exclude=.git' '--exclude=*~' --recursive -U 5 --new-file uClibc-0.9.32.1-org/libc/sysdeps/linux/xtensa/bits/kernel_types.h uClibc-0.9.32.1/libc/sysdeps/linux/xtensa/bits/kernel_types.h
+--- uClibc-0.9.32.1-org/libc/sysdeps/linux/xtensa/bits/kernel_types.h	2011-12-30 03:11:07.000000000 -0800
++++ uClibc-0.9.32.1/libc/sysdeps/linux/xtensa/bits/kernel_types.h	2014-07-17 14:51:52.447540298 -0700
+@@ -31,10 +31,12 @@
+ typedef unsigned int	__kernel_gid32_t;
+ 
+ typedef unsigned short	__kernel_old_uid_t;
+ typedef unsigned short	__kernel_old_gid_t;
+ typedef unsigned short	__kernel_old_dev_t;
++typedef long		__kernel_long_t;
++typedef unsigned long	__kernel_ulong_t;
+ typedef long long	__kernel_loff_t;
+ 
+ /* Beginning in 2.6 kernels, which is the first version that includes the
+    Xtensa port, __kernel_dev_t is defined in "linux/types.h" and is no longer
+    architecture-specific.  It is defined here in uClibc for consistency with
